code/modules/admin/debug.dm

```diff
@@ -1346,3 +1346,21 @@ var/datum/flock/testflock
 
 
 	boutput(usr, "<span class='notice'>Returned: [!isnull(returnval) ? returnval : "null"]</span>")
+
+/proc/debugRemoveComponent(var/datum/target = null)
+	var/list/dc = target.datum_components
```

Comment:
untyped list?

Reply:
datum_components is present on every datum and it is a regular untyped list.